articles/api-auth/tutorials/verify-access-token.md

```diff
@@ -0,0 +1,135 @@
+
+We should note here that many web frameworks (such as [ASP.NET Core](/quickstart/backend/aspnet-core-webapi) for example) have JWT middleware that handle the token validation. Most of the times this will be a better route to take, rather that resorting to use a third-party library, as the middleware typically integrates well with the framework's overall authentication mechanisms.
+
+### How can I visually inspect a token?
+
+A quick way to see what is inside a JWT is by using the [JWT.io](https://jwt.io/) website (alternatively, you can use the [JWT Debugger Chrome Extension](https://chrome.google.com/webstore/detail/jwt-debugger/ppmmlchacdbknfphdeafcbmklcghghmd?hl=en)). It has a handy debugger which allows you to quickly check that a JWT is well formed, and also inspect the values of the various claims.
+
+Just paste your token at the _Encoded_ text area and review the decoded results at the right.
+
+![Decode JWT with JWT.io](/media/articles/api-auth/decode-jwt.png)
+
+## Check the Signature Algorithm
+
+The API needs to check if the algorithm, as specified by the JWT header (property `alg`), matches the one expected by the API. If not, the token is considered invalid and the request must be rejected.
+
+In this case the mismatch might be due to mistake (it is common that the tokens are signed using the `HS256` signing algorithm, but your API is configured for `RS256`, or vice versa), but it could also be due to an attack, hence the request has to be rejected.
+
+### How can I check the signature algorithm?
+
+To check if the signature matches the API's expectations, you have to decode the JWT and retrieve the `alg` property of the JWT header.
+
+Alternatively, you can use one of the libraries listed in the _Libraries for Token Signing/Verification_ section of [JWT.io](https://jwt.io/).
+
+Following the Node.js example of the previous section, the [jwt.verify()](https://github.com/auth0/node-jsonwebtoken#jwtverifytoken-secretorpublickey-options-callback) method of the [node-jsonwebtoken library](https://github.com/auth0/node-jsonwebtoken), supports an `algorithms` argument, that contains a list of strings with the names of the allowed algorithms.
+
+## Verify the Signature
+
+The API needs to verify the signature. The signature is used to verify that the sender of the JWT is who it says it is and to ensure that the message wasn't changed along the way.
+
+Remember that the signature is created using the header and the payload of the JWT, a secret and the hashing algorithm being used (as specified in the header: HMAC, SHA256 or RSA). The way to verify it, depends on the hashing algorithm:
+- For `HS256`, the Client's Secret is used. You can find this information at your [Client's Settings](https://${manage_url}/#/clients/${account.clientId}/settings).
```

Comment:
The API secret is used, not the client's.

![image](https://cloud.githubusercontent.com/assets/1980994/23034155/8f78d2bc-f459-11e6-96b4-32d71812cb92.png)